recipe/build.sh

```diff
@@ -12,8 +15,16 @@ else
       export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0;7.5;8.0"
   elif [[ ${cuda_compiler_version} == 11.1* ]]; then
       export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0;7.5;8.0;8.6"
-  elif [[ ${cuda_compiler_version} == 11.2* ]]; then
-      export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0;7.5;8.0;8.6"
+  elif [[ "$cuda_compiler_version" == "11.2*" ]]; then
+    export CUDA_ARCH_LIST="${CUDA_ARCH_LIST},sm_35"
+    export CUDAARCHS="${CUDAARCHS};35-virtual;80-virtual"
+    export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0;7.5;8.0;8.6"
+  elif [[ "$cuda_compiler_version" == "11.8" ]]; then
+    export CUDA_ARCH_LIST="${CUDA_ARCH_LIST},sm_35,sm_89,sm_90"
+    export CUDAARCHS="${CUDAARCHS};35-virtual;89-real;90"
+  elif [[ "$cuda_compiler_version" == "12.0" ]]; then
+    export CUDA_ARCH_LIST="${CUDA_ARCH_LIST},sm_89,sm_90"
+    export CUDAARCHS="${CUDAARCHS};89-real;90"
```

Comment:
Since the rest of these are now up top, we just need to add `3.5`, which is conditionally supported

Also would stick to `TORCH_CUDA_ARCH_LIST` as that is all that is relevant to us (the rest are MAGMA specific)

```suggestion
    export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;8.9;9.0"
```